### PR TITLE
Working on search bar

### LINKED ISSUE #6: Add search functionality
Adding search bar at the top of the table. It should show todos by search values.

### PR DESCRIPTION
**Pull request by dgupta0**
fixes #6

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
-import { Table, Tag } from 'antd';
+import { Table, Tag, Input } from 'antd';
 
 
+const { Search } = Input;
+
 function App() {
   const [data, setData] = useState(null);
+  const [filterVal, setFilterVal] = useState("")
+  const [filteredData, setFilteredData] = useState(null);
```
Comment (Dorfieeee) [inline src/App.jsx:11]: I think that empty array is a better solution. You don't have to do checks like `data || []` later in the code every time...

```suggestion
  const [filteredData, setFilteredData] = useState([]);
```

File: src/App.jsx [hunk 2]
```diff
@@ -80,12 +115,59 @@ function App() {
...
+  function handleSearch() {
+    if (!filterVal) {
+      setFilteredData(data)
+    } else {
+      setFilteredData(
+        (data || []).filter(todo =>
+          todo.title.toLowerCase()
+            .includes(filterVal.toLowerCase())
+        )
+      )
```
Comment (Dorfieeee) [inline src/App.jsx]: What is stopping you here from doing the checks for all the todo properties (`"timeStamp", "dueDate", "tags", "status", "description", "title"`) ???
When you add more checks for the other properties you can expand the search to all todo's fields just as originally required.

Comment (dgupta0) [reply src/App.jsx]: I just want to keep the search simple right now. And we both decided that for now we will only implement search by title.

File: src/App.jsx (hunk 1 above)
Comment (dgupta0) [reply src/App.jsx:11]: makes sense. I didn't realise that before.

**Pull request merged**